File: server/hotrod/src/main/scala/org/infinispan/server/hotrod/ErrorResponse.scala
package org.infinispan.server.hotrod

/**
 * // TODO: Document this
 * @author Galder Zamarreño
 * @since 4.1
 */

class ErrorResponse(override val id: Long,
                    override val opCode: OpCodes.OpCode,
                    override val status: Status.Status,
                    val msg: String) extends Response(id, opCode, status) {

   override def toString = {
      new StringBuilder().append("ErrorResponse").append("{")
         .append("id=").append(id)
         .append(", opCode=").append(opCode)
         .append(", status=").append(status)
         .append(", msg=").append(msg)
         .append("}").toString
   }
   
}

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/ErrorResponse.java
package org.infinispan.server.hotrod;

import org.infinispan.server.hotrod.transport.ExtendedByteBuf;

import io.netty.buffer.ByteBuf;

/**
 * @author wburns
 * @since 9.0
 */
public class ErrorResponse extends Response {
   protected final String msg;

   ErrorResponse(byte version, long messageId, String cacheName, short clientIntel, OperationStatus status, int topologyId, String msg) {
      super(version, messageId, cacheName, clientIntel, HotRodOperation.ERROR, status, topologyId);
      this.msg = msg;
   }

   public String getMsg() {
      return msg;
   }

   @Override
   public String toString() {
      return "ERROR{" +
            "version=" + version +
            ", messageId=" + messageId +
            ", cacheName='" + cacheName + '\'' +
            ", clientIntel=" + clientIntel +
            ", operation=" + operation +
            ", status=" + status +
            ", topologyId=" + topologyId +
            ", msg=" + msg +
            '}';
   }
}
